### PR TITLE
licensing: display license validity status in site-admin license page

### PR DESCRIPTION
**Pull request by BolajiOlajide**
This PR updates the UI for the License page `/site-admin/license` so it displays more information about the validity of a license key.

#### Before

CleanShot 2023-07-14 at 13 10 39@2x

#### After

CleanShot 2023-07-14 at 13 09 54@2x

## Test plan

Manually updating the values of:

* `licensing:is_license_valid`
* `licensing:license_invalid_reason`

The first key should reflect the validity status of the Sourcegraph license key, while the second should return the reason it's invalid (if it is).

@sourcegraph/design I appreciate some feedback on the design here.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/authz/init.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func Init(
 
 		return []*graphqlbackend.Alert{{
 			TypeValue:    graphqlbackend.AlertTypeError,
-			MessageValue: fmt.Sprintf("The Sourcegraph license key is invalid. Reason: %s. To continue using Sourcegraph, a site admin must renew the Sourcegraph license (or downgrade to only using Sourcegraph Free features). Update the license key in the [**site configuration**](/site-admin/configuration). Please contact Sourcegraph support for more information.", reason),
+			MessageValue: fmt.Sprintf("The Sourcegraph license key is invalid. Reason: %q. To continue using Sourcegraph, a site admin must renew the Sourcegraph license (or downgrade to only using Sourcegraph Free features). Update the license key in the [**site configuration**](/site-admin/configuration). Please contact Sourcegraph support for more information.", reason),
```
Comment (BolajiOlajide) [inline enterprise/cmd/frontend/internal/authz/init.go:90]: Quoting this string makes it easier to read, In my opinion.

@kopancek WDYT?

**Review by kopancek — APPROVED** [review]
LGTM :shipit:

File: enterprise/cmd/frontend/internal/authz/init.go (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/authz/init.go:90]: Personal taste, but in the screenshot I find it harder to read, somewhat :D

File: client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx [hunk 2]
```diff
@@ -191,3 +182,35 @@ export const ProductSubscriptionStatus: React.FunctionComponent<React.PropsWithC
...
+                    showPrefix={true}
+                />
+            </>
+        )
+    }
+
+    return (
+        <Alert variant="danger">
+            <Text className="mb-0">
+                The Sourcegraph license key is invalid. Reason: "{license.licenseInvalidityReason}"
```
Comment (eseliger) [inline client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx:212]: do we have to quote this?

**Review by eseliger — APPROVED** [review]

File: client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx (hunk 2 above)
Comment (BolajiOlajide) [reply client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx:212]: Makes it standout, In my opinion. 

To answer your question, we don't.

File: enterprise/cmd/frontend/internal/authz/init.go (hunk 1 above)
Comment (BolajiOlajide) [reply enterprise/cmd/frontend/internal/authz/init.go:90]: Yeah, I've gotten some feedback from Paulo to modify this.

File: client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx (hunk 2 above)
Comment (BolajiOlajide) [reply client/web/src/enterprise/site-admin/productSubscription/ProductSubscriptionStatus.tsx]: REmoved the quotes

**Pull request merged**